### DISCUSSION TITLE
error: The signal ... does not fit in message ...

### DISCUSSION THREAD

**Discussion by msawye16** [cantools/cantools#473] [Q&A]
I have recently encountered DBC files that have signals which extend beyond 8-bytes.

The offending message is actually a CAN FD message which is received through a gateway.

I have been told that the message, if seen on the local bus, will never be more than 8-bytes.

My problem is that `cantools` complains with an error message and will not proceed any further when dumping or listing the DBC file.

Is there a way that I can have `cantools` filter out and ignore messages with issues during processing?

**Comment by juleq** [comment]
You could try removing the offending message definition from the dbc or alter it in a way that makes it a legal non fd message.

**Reply by andlaus** [reply]
alternatively, you can try if the `--non-strict` option helps. It is available for most (all?) subparsers:
```
cantools dump --non-strict $FILE.dbc
```

**Reply by msawye16** [reply]
There does not appear to be any parsing for `-non-strict` at the point where the exception occurs so that would not work for me.

My solution is an inelegant one but it allows me to continue my real work :wink: 
I simply display an error message and ignore the offending CAN message during parsing.

![image](https://user-images.githubusercontent.com/33455205/186418010-6f999949-82e2-4daa-a7fe-fc9f196faad5.png)
